src/main/java/ip/view/MainWindow.java

```diff
@@ -0,0 +1,65 @@
+package ip.view;
+
+import ip.Duke;
+import ip.Ui;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Duke duke;
+
+    private Image leftImage = new Image(this.getClass().getResourceAsStream("/images/left_red.png"));
+    private Image rightImage = new Image(this.getClass().getResourceAsStream("/images/right_red.png"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        dialogContainer.getChildren().add(DialogBox.getRightDialog(Ui.initialGreeting(), rightImage));
+    }
+
+    public void setDuke(Duke d) {
```

Comment:
While understandable here, single char variables or parameters can be bad habits in the long run. Maybe try more descriptive names to avoid confusion in the future.